shopgate/pwa
# Implemented automatic logout on session expiry

# Description

This ticket is about to implement an automatic logout feature after a user session expired. Therefore it uses the `sessionLifetimeInSeconds` property that comes with a login response when the extension setting is configured.

When active the system periodically checks if the session is expired and logs the user out when it happens. In that case users will see a special dialog that describes what just happened.

## Type of change


## How to test it

Attach the `@shopgate/connect-trusted` extension and configure the `sessionLifetimeInSeconds` extension setting. Start the frontend, log in and check what happens when the session expires. Check how it performs when the app is active while the session expires, also check how it performs when the app is in the background when the session expires.
